### PR TITLE
Fix logger dependencies

### PR DESCRIPTION
**Pull request by MathewMacDougall**
### Description

Noticed after Kelvin's PR we weren't depending on "our" logger most of the time and skipping right to g3log. I've changed everything to depend on our logger rather than `g3log` for consistency (and correctness).

### Testing Done

Code still builds

### Resolved Issues

none

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/external/g3log.BUILD [hunk 1]
```diff
@@ -43,5 +43,5 @@ cc_library(
         "./include",
         "./src",
     ],
-    visibility = ["//visibility:public"],
+    visibility = ["@//software/logger:__subpackages__"],
```
Comment (garethellis0) [inline src/external/g3log.BUILD:46]: Very nice :slightly_smiling_face:

**Review by garethellis0 — APPROVED** [review]
It's worth noting here that we still have an implicit dependency on G3log because we call it's macros directly all over the codebase (ex. `LOG(WARNING) << "blah blah";`) but this is certainly an improvement.

**Review by jonathanlew — APPROVED** [review]
Nice find!

**Comment by MathewMacDougall** [comment]
Yup I'm aware of the implicitly dependency, but this should at least be more consistent and make it easier to change loggers should we ever need to.

**Review by kelvinkoon — APPROVED** [review]
LGTM! :+1:

**Pull request merged**